### PR TITLE
Loading and saving of single value ws

### LINKED ISSUE #34021: SingleValued workspaces not saved with the project
An ILL user has reported that single valued workspaces created as part of their reduction workflow, and containing crucial information such as initial neutron energy, are not saved along with the project. This naturally causes a major issue when the project is loaded later, and this relevant information is missing, effectively making the saved project unusable.

**Describe the bug**

Single value workspaces are not saved with the project, this of workspace is not currently supported by `SaveNexus` or `SaveNexusProcessed`. 

**To Reproduce**

Single value workspaces should be properly saved and loaded, and present in the project that is saved by a user.

**Screenshots**

**Platform/Version (please complete the following information):**
 - OS: [e.g. Windows, RHEL 7, Ubuntu, macOS]
 - OS Version:
 - Mantid Version [e.g. 6.0.0]

**Additional context**

### PR DESCRIPTION
**Pull request by darominski**
**Description of work.**

This PR enables saving and loading of single value workspaces (`WorkspaceSingleValue`) to address the found issue where this type of workspaces would not be saved with the project, causing issues in recuperating reduction workflow started in a separate session. During loading, the proposed differentiation between a `Workspace2D` that has one bin and one spectrum from the special type of `WorkspaceSingleValue` is that if a workspace to be loaded contains no metadata (except for the default group included, called 'goniometer') AND has no instrument attached (instrument being an empty string), then it is effectively a single value workspace.

This PR contains also contributions from @MathieuTillet . 

**To test:**

1. Run the following script:
```
from mantid.simpleapi import *
import numpy as np

sv_ws = 'singlevalue_ws'
CreateSingleValuedWorkspace(15, OutputWorkspace=sv_ws)

sv_like_ws = '2dws_w_singlevalue'
CreateWorkspace(DataX=np.array([0]), DataY=np.array([5]), NSpec=1, OutputWorkspace=sv_like_ws)

print(type(mtd[sv_ws]))      # this should be WorkspaceSingleValue
print(type(mtd[sv_like_ws])) # this should be Workspace2D
path1=f'{sv_ws}_nxps.nxs'    # nexus processed
SaveNexusProcessed(InputWorkspace=sv_ws, Filename=path1)
path2=f'{sv_ws}_nxs.nxs' # nexus
SaveNexus(InputWorkspace=sv_ws, Filename=path2)
path3=f'{sv_like_ws}_nxps.nxs' # nexus processed
SaveNexusProcessed(InputWorkspace=sv_like_ws, Filename=path3)
path4=f'{sv_like_ws}_nxs.nxs' # nexus
SaveNexus(InputWorkspace=sv_like_ws, Filename=path4)

LoadNexus(Filename=path1, OutputWorkspace=f'{sv_ws}_nxps_loaded')
LoadNexus(Filename=path2, OutputWorkspace=f'{sv_ws}_nxs_loaded')
LoadNexus(Filename=path3, OutputWorkspace=f'{sv_like_ws}_nxps_loaded')
LoadNexus(Filename=path4, OutputWorkspace=f'{sv_like_ws}_nxs_loaded')

print(type(mtd[f'{sv_ws}_nxps_loaded']))      # should be still WorkspaceSingleValue
print(type(mtd[f'{sv_ws}_nxs_loaded']))       # should be still WorkspaceSingleValue
print(type(mtd[f'{sv_like_ws}_nxps_loaded'])) # should be still Workspace2D
print(type(mtd[f'{sv_like_ws}_nxs_loaded']))  # should be still Workspace2D
```

2. There should be no error mentioned in the issue #34021 , and the types should agree with the comments next to the line.
3. Save the project, and clear all workspaces in the ADS
4. Load the saved project: all workspaces previously present in the ADS should be properly loaded and the single valued workspace should not be missing and should be of the proper type.

Fixes #34021. 

Release notes added.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/DataHandling/src/LoadNexusProcessed.cpp [hunk 1]
```diff
@@ -778,7 +778,7 @@ void LoadNexusProcessed::loadNumericColumn(const Mantid::NeXus::NXData &tableDat
 /**
  * Load a table
  */
-API::Workspace_sptr LoadNexusProcessed::loadTableEntry(NXEntry &entry) {
+API::Workspace_sptr LoadNexusProcessed::loadTableEntry(const NXEntry &entry) {
```
Comment (Harrietbrown) [inline Framework/DataHandling/src/LoadNexusProcessed.cpp:781]: This needs to be made const in the heder as well.

**Comment by darominski** [comment]
For some reason, an instrument created using `std::make_shared<Instrument>(name)` differs when called from Python API, which connects to C++ `CreateSampleWorkspace` and the test for `LoadNexusProcessed2`, which connects to `ComponentCreationHelper`. The test is failing because in the processed nexus a group named after the instrument (e.g. 'basic_rect') is created instead of a group called 'instrument' containing all of the details of the attached instrument. I am looking into this behaviour discrepancy.

**Comment by darominski** [comment]
My previous comment was linked with the different approach to store the instrument data by the `SaveNexusESS` algorithm, which puts all the information in a group with the name of the instrument, rather than in a group called 'instrument', like in `SaveNexusProcessed-v1`. The proposed solution is not the prettiest, but addresses this situation. I am open to alternative solutions, do let me know if you have any suggestions.

**Comment by darominski** [comment]
@Harrietbrown Please let me know if you have any other comments or change requests regarding this PR.

**Review by Harrietbrown — APPROVED** [review]
Saving and loading worksapces works as expected, and the project save is also working. good job.

**Pull request merged**